evap/staff/views.py

```diff
@@ -91,6 +90,100 @@ def semester_view(request, semester_id):
 
 
 @staff_required
+def semester_course_operation(request, semester_id):
+    semester = get_object_or_404(Semester, id=semester_id)
+    raise_permission_denied_if_archived(semester)
+
+    course_ids = request.GET.getlist('course')
+    operation = request.GET.get('operation')
+    if operation not in ['revertToNew', 'prepare', 'reenableLecturerReview', 'approve', 'publish', 'unpublish']:
+        messages.error(request, _("Unsupported operation: ") + str(operation))
+        return custom_redirect('evap.staff.views.semester_view', semester_id)
+    courses = [get_object_or_404(Course, id=course_id) for course_id in course_ids]
```

Comment:
i think a simple get (and thus a crash if the course does not exist) is more appropriate.